js/post-edit.js

```diff
@@ -42,8 +44,11 @@ function WPSimplechartApp(){
 				app.modalElements.container = app.modalElements.container.replace('{{iframeSrc}}', WPSimplechartBootstrap.appUrl);
 				app.modalElements.container = app.modalElements.container.replace('{{closeModal}}', WPSimplechartBootstrap.closeModal);
 				$( 'body' ).append( app.modalElements.container + app.modalElements.backdrop );
-				$( '#simplechart-close' ).click( function(){
-					$( '#simplechart-backdrop, #simplechart-modal' ).hide();
+				$( '#simplechart-close' ).click( function( e ) {
+					e.preventDefault();
+					if ( app.chartData || confirm( app.confirmNoDataMessage ) ) {
```

Comment:
assuming `app.chartData` is only populated by **Save Chart**. It gets reset eg if you reopen the app modal?